README.md

```diff
@@ -1,85 +1,236 @@
+    ```bash
+    cp bin/pulumi-resource-foo $GOPATH/bin
+    ```
 
-    $ npm install @pulumi/xyx
+1. Tell Yarn to use your local copy of the SDK:
 
-or `yarn`:
+    ```bash
+    make install_nodejs_sdk
+    ```
 
-    $ yarn add @pulumi/xyx
+1. Create a new Pulumi program in the `examples/` directory, e.g.:
 
-### Python
+    ```bash
+    mkdir examples/my-example/ts # Change "my-example" to something more meaningful.
+    cd examples/my-example/ts
+    pulumi new typescript
+    # (Go through the prompts with the default values)
+    npm install
+    yarn link @pulumi/foo
+    ```
 
-To use from Python, install using `pip`:
+1. Create a minmal program for the provider, i.e. one that creates the smallest-footprint resource.  Place this code in `index.ts`.
+1. Configure any necessary environmental variables and ensure the program runs successfully via `pulumi up`.
+1. Once the program completes successfully, verify the resource was created in the provider's UI.
+1. Destroy any resources created by the progam via `pulumi destroy`.
 
-    $ pip install pulumi_xyx
+Optionally, you may create additional examples for SDKs in other languages supported by Pulumi.
 
-### Go
+## Configuring CI/CD with GitHub Actions
 
-To use from Go, use `go get` to grab the latest version of the library
+In this section, we'll add the necessary configuration to work with GitHub Actions for Pulumi's standard CI/CD workflows for providers.
 
-    $ go get github.com/pulumi/pulumi-xyz/sdk/go/...
+1. Per the [README in the `ci-mgmt` repo](https://github.com/pulumi/ci-mgmt/#readme), generate the necessary GitHub Actions files and place them in the `.github/actions` directory in this repository.
+1. Add code to `examples_nodejs_test.go` to call the example you created, e.g.:
 
-## Configuration
+    ```go
+    // Swap out MyExample and "my-example" below with the name of your integration test.
+    func TestAccMyExampleTs(t *testing.T) {
+        test := getJSBaseOptions(t).
+            With(integration.ProgramTestOptions{
+                Dir: filepath.Join(getCwd(t), "my-example", "ts"),
+            })
+        integration.ProgramTest(t, &test)
+    }
+    ```
 
-The following configuration points are available for the `xyz` provider:
+    Add a similar function for each example that you want to run in an integration tests.  For examples written in other languages, create similar files for `examples_${LANGUAGE}_test.go`.
+1. Ensure that any required secrets (API keys, etc.) are present in GitHub.
 
-- `xyz:apiKey` (environment: `XYZ_API_KEY`) - the API key for `xyz`
-- `xyz:region` (environment: `XYZ_REGION`) - the region in which to deploy resources
+## Final Steps
 
-## Reference
+1. Replace this file with the README for the provider and create a PR to merge the changes:
 
-For detailed reference documentation, please visit [the API docs][1].
+    ```bash
+    mv README README-BIOLERPLATE.md # README-BOILERPLATE.md is .gitignored by default.
```

Comment:
This feels awkward - why would they keep the boilerplate README?

Reply:
I guess my thinking was that it's a long process and one could easily make a mistake, although the boilerplate repo is obviously there for the user to refer to if we directly overwrite the boilerplate README.

Reply:
Removed.  Marking resolved.